src/skmultiflow/data/temporal_data_stream.py

```diff
@@ -0,0 +1,148 @@
+import pandas as pd
+import numpy as np
+
+import warnings
+
+from skmultiflow.data.data_stream import DataStream
+
+# implement temporal data stream
+class TemporalDataStream(DataStream):
+    """ Creates a temporal stream from a data source.
+
+    TemporalDataStream takes the whole data set containing the `X` (features), `time` (timestamps) and `Y` (targets).
+
+    Parameters
+    ----------
+    data: np.ndarray or pd.DataFrame (Default=None)
+        The features' columns and targets' columns or the feature columns
+        only if they are passed separately.
+    time: np.ndarray(dtype=datetime64) or pd.DataFrame (Default=None)
+        The timestamp column of each instance. If its a np.ndarray, it will
+        be converted into a pandas datetime dataframe. 
+    sample_weight: np.ndarray or pd.DataFrame, optional (Default=None)
+        Sample weights.
+    sample_delay: np.ndarray(pd.tseries.offsets.DateOffset) or pd.tseries.offsets.DateOffset, optional (Default=pd.tseries.offsets.DateOffset(day=0))
+        Samples delay in pd.tseries.offsets.DateOffset (the dateoffset difference between the event time 
+        and when the label is available).
+    y: np.ndarray or pd.DataFrame, optional (Default=None)
+        The targets' columns.
+    target_idx: int, optional (default=-1)
+        The column index from which the targets start.
+
+    n_targets: int, optional (default=1)
+        The number of targets.
+
+    cat_features: list, optional (default=None)
+        A list of indices corresponding to the location of categorical features.
+
+    name: str, optional (default=None)
+        A string to id the data.
+
+    ordered: bool, optional (default=True)
+        If True, consider that data, time and y are already ordered by timestamp.
+        Otherwise, the data is ordered based on `time` timestamps.
+
+    allow_nan: bool, optional (default=False)
+        If True, allows NaN values in the data. Otherwise, an error is raised.
+
+    Notes
+    -----
+    The stream object provides upon request a number of samples, in a way such that old samples cannot be accessed
+    at a later time. This is done to correctly simulate the stream context.
+
+    """
+    # includes time as datetime
+    def __init__(self, data, time, y=None, sample_weight=None, sample_delay=pd.tseries.offsets.DateOffset(day=0), target_idx=-1, n_targets=1, cat_features=None, name=None, ordered=True):
```

Comment:
According to docstring, `data` is also `None`. What should we do?

Reply:
By the way, the same happen in `data_stream.py`.

Reply:
You re right. If `data` is None it will lead to an exception. See below https://github.com/scikit-multiflow/scikit-multiflow/blob/92f7ac63efd89d1bbe27707cab35f3be452aedfa/src/skmultiflow/data/data_stream.py#L66-L67

`data` is a required parameter. We need to check if the `data` is not None otherwise, we raise an exception.  
Let's handle it here and fix the others later in different PR.  

Could you please update the docstring?

Reply:
> You re right. If `data` is None it will lead to an exception. See below
> 
> https://github.com/scikit-multiflow/scikit-multiflow/blob/92f7ac63efd89d1bbe27707cab35f3be452aedfa/src/skmultiflow/data/data_stream.py#L66-L67
> 
> `data` is a required parameter. We need to check if the `data` is not None otherwise, we raise an exception.
> Let's handle it here and fix the others later in different PR.
> 
> Could you please update the docstring?

Updated in commit [3a8f283c592955b4f6e05228953c5fc3da6f80d8](https://github.com/scikit-multiflow/scikit-multiflow/commit/3a8f283c592955b4f6e05228953c5fc3da6f80d8).